SeaPublicWebsite/Services/EnergyEfficiency/Epc/EpcFilterService.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using SeaPublicWebsite.BusinessLogic.ExternalServices.EpbEpc;
+using SeaPublicWebsite.BusinessLogic.Models;
+
+namespace SeaPublicWebsite.Services.EnergyEfficiency.Epc;
+
+public class EpcFilterService
+{
+    private readonly IEpcApi epcApi;
+
+    public EpcFilterService(IEpcApi epcApi)
+    {
+        this.epcApi = epcApi;
+    }
+    
+    public async Task<List<EpcSearchResult>> FilterEpcsNotUniqueByAddressToMostRecent(IEnumerable<EpcSearchResult> searchResults)
+    {
+        
+        var epcSearchResultsGroupedByAddress = searchResults.GroupBy(
+            epcSearchResult => new
+            {
+                epcSearchResult.Address1, 
+                epcSearchResult.Address2, 
+                epcSearchResult.Postcode
+            });
+
+
+        var epcSearchResultsNotUniqueByAddress = epcSearchResultsGroupedByAddress
+            .Where(grouping => grouping.Count() > 1)
+            .SelectMany(grouping => grouping);
+        var epcSearchResultsUniqueByAddress = epcSearchResultsGroupedByAddress
+            .Where(grouping => grouping.Count() == 1)
+            .SelectMany(grouping => grouping);
+            
+        var epcSearchResultsWithAdditionalEpcInformationTaskList = epcSearchResultsNotUniqueByAddress.Select(async epc => new KeyValuePair<EpcSearchResult, BusinessLogic.Models.Epc>(epc, await epcApi.GetEpcForId(epc.EpcId)));
+
+        var epcSearchResultsWithAdditionalEpcInformation = await Task.WhenAll(epcSearchResultsWithAdditionalEpcInformationTaskList);
+
+        var epcSearchResultsNotUniqueByAddressWithOlderDuplicatesRemoved = epcSearchResultsWithAdditionalEpcInformation
+            .Where(epcDtoPair => epcDtoPair.Value.IsLatestAssessmentForAddress)
+            .Select(epcDtoPair => epcDtoPair.Key);
```

Comment:
Might just be me not understanding how EPCs work in the codebase, but `IsLatestAssessmentForAddress` is nullable and we're not doing anything to populate it in this PR. Until the field is correctly populated for each address we risk returning nothing when there are several results.
Is that actually a risk?

Reply:
Just in case someone is looking at this for tech debt. From how I understood it (and seemed to be the case testing locally). IsLatestAssessmentForAddress was part of the object returned by the EPC Api. 

Essentially the search by address endpoint returns a list of EPC ids with a little information on them (but not the full object). The bug is caused by this endpoint sometimes returning multiple for the same address.

The EPC by id endpoint returns a more complete object which includes `IsLatestAssessmentForAddress`. If we get 2 EPCs for an address from the search endpoint we check the full EPCs in this function. (That being said I suppose there could be a weird case where neither of the search EPCs are the latest. That might be worth thinking about).